gcloud/storage/bucket.py

```diff
@@ -222,14 +227,22 @@ def get_blob(self, blob_name, client=None):
         :param client: Optional. The client to use.  If not passed, falls back
                        to the ``client`` stored on the current bucket.
 
+        :type generation: int
+        :param generation: Optional. The generation id to retrieve in a bucket
+                           that supports versioning.
+
         :rtype: :class:`gcloud.storage.blob.Blob` or None
         :returns: The blob object if it exists, otherwise None.
         """
         client = self._require_client(client)
         blob = Blob(bucket=self, name=blob_name)
         try:
+            path = blob.path
+            if generation is not None:
+                path = (blob.path + '?generation=' +
+                        quote(str(generation), safe=''))
```

Comment:
I did this originally.  But, after seeing how the code was coming together and thinking of the models a bit, I came to the conclusion that the bucket was a better place for this.  

Here is my reasoning; I'd love your feedback:

1) The blob object doesn't facilitate querying, rather it's a representation of what the storage system believes exists.  Allowing someone to define the generation of a blob, through a setter, seems to be counter to the purpose of the generation property on the blob.  That is, blob.generation is the canonical ID returned by the storage system for that blob.  Not something a user should be able to define.  

2) The blob.path is an imputed property and constructed from the only two key pieces of information needed to define a blob: bucket and name.  User-defined generation seems a bit superfluous, since it has no real meaning in the definition of a blob.

These two points led me to think of the generation as a query parameter better suited to the bucket API.

The problem with my approach now is that it disregards all other query parameters (ifGenerationNotMatch, ifMetagenerationNotMatch, etc.).  

I think the next step for this path of development would be a generator function, in the bucket, that uses user-specified query parameters and the blob.path to create the full API query URL.

Let me know your thoughts... if you would like the query param to be through a setter to be at the blob level, and corresponding blob.path construction to happen there, I can refactor.